nemo/collections/nlp/models/language_modeling/megatron_base_prompt_learning_model.py

```diff
@@ -296,26 +212,33 @@ def load_state_dict(self, state_dict, strict: bool = True):
         parameters. Matching load method for this class' custom state dict method. 
         """
         if self.first_stage_of_pipeline():
-            if self._prompt_table_key in state_dict:
-                state_dict_ = state_dict[self._prompt_table_key]
+            if self.virtual_prompt_source == VirtualPromptSource.PROMPT_ENCODER:
+                if self.prompt_encoder is None:
+                    self.init_prompt_encoder()
+                self.prompt_encoder.load_state_dict(state_dict, strict)
             else:
-                # Handle loading state dict before weight saving change for backward compatibility.
-                state_dict_ = OrderedDict()
-                for key in state_dict.keys():
-                    if key.startswith(self._prompt_table_key):
-                        key_substring = ".".join(key.split(".")[1:])
-                        state_dict_[key_substring] = state_dict[key]
-
-            self.prompt_table.load_state_dict(state_dict_, strict)
-
-            if (
-                self._prompt_encoder_key in state_dict
-                and self.virtual_prompt_source == VirtualPromptSource.PROMPT_ENCODER
-            ):
-                state_dict_ = state_dict[self._prompt_encoder_key]
-                self.prompt_encoder.load_state_dict(state_dict_, strict)
-
-    def embed_input_train(self, input_ids: Tensor, taskname_ids: Tensor):
+                raise ValueError("invalid virtual prompt source")
```

Comment:
please indicate what is the correct config. E.g. please set this cfg.... to p_tuning